src/cryptography/x509.py

```diff
@@ -1158,3 +1225,18 @@ def signature_hash_algorithm(self):
         Returns a HashAlgorithm corresponding to the type of the digest signed
         in the certificate.
         """
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RevokedCertificate(object):
```

Comment:
Sorry, I got two things confused. This has nothing to do with verify(), this will work correctly.
What I meant was that I originally had a few additional convenience functions called is_revoked() and get_revoked() which would take a cert as an argument so you can check if a certain cert is actually revoked. Since revocation is done by serial number you have to make sure the cert actually matches the CRL, for which you compare issuers.
I'm just leaving these functions out so a user would have to do the logic themselves.

So no, the cryptography library would not be doing anything wrong, it just might not provide the interfaces you need.